File: 7.HammingDistance/HammingDistance.py


def HammingDistance(string1, string2):
    return sum(1 for x, y in zip(string1, string2) if x != y)


def main():
    string1 = "GATAGCAGCTTCTGAACTGGTTACCTGCCGTGAGTAAATTAAAATTTTATTGACTTAGGTCACTAAATACT"
    string2 = "AATAGCAGCTTCTCAACTGGTTACCTCGTATGAGTAAATTAGGTCATTATTGACTCAGGTCACTAACGTCT"
    print(HammingDistance(string1, string2))


if __name__ == "__main__":
    main()

File: 7.HammingDistance/HammingDistance.go
package main

import "fmt"

func HammingDistance(a, b string) int {
	if len(a) != len(b) {
		fmt.Println("Strings must have the same size")
		return 1
	}
	dist := 0
	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			dist++
		}
	}
	return dist
}

func main() {
	a := "GATAGCAGCTTCTGAACTGGTTACCTGCCGTGAGTAAATTAAAATTTTATTGACTTAGGTCACTAAATACT"
	b := "AATAGCAGCTTCTCAACTGGTTACCTCGTATGAGTAAATTAGGTCATTATTGACTCAGGTCACTAACGTCT"
	res := HammingDistance(a, b)
	fmt.Println(res)
}
